### DISCUSSION TITLE
How do I specify the -D option or --recurse option when using ikvmc.exe directly when building with dotnet build using IkvmReference?

### DISCUSSION THREAD

**Discussion by kdaek21** [ikvmnet/ikvm#617] [Q&A]
As the title says.

How do I specify the -D option or --recurse option when using ikvmc.exe directly when building with dotnet build using IkvmReference?

**Comment by wasabii** [comment]
Neither of those options make much sense there. 

-D is for setting system properties. IkvmReference is for adding libraries as references. Setting system properties on a library doesn't make sense. 

Recurse just allows you to specify a wildcard. But this is MSBuild where wildcards are built-in.

**Comment by kdaek21** [comment]
Additional question.

How can I bundle two or more jars into one assembly instead of linking them by reference?

Like [classOrJar1] ...[classOrjarN] part of ikvmc.exe..

**Reply by wasabii** [reply]
Set the Compile metadata as described in README.

**Reply by kdaek21** [reply]
I also found a way to insert Java properties (I found that you can insert them in the C# class that acts as a wrapper in the format java.lang.System.setProperty("variable name", "value");), and after asking additional questions and testing various things, I found a way to insert Compile metadata, so I will close that question.

**Reply by wasabii** [reply]
Sure. Do keep on mind that just like in Java, there are some properties which can only be set before the runtime initializes.
